### PR TITLE
ramips: some fixes for HiWiFi HC5761A

### PR DESCRIPTION
**Pull request by dengqf6**
Fix switch ports configuration
Fix USB port power

### CODE REVIEW

File: target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts [hunk 1]
```diff
@@ -32,11 +32,22 @@
 			linux,default-trigger = "phy1tpt";
 		};
 	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		usbpower {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: usb_power or power_usb

File: target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts [hunk 2]
```diff
@@ -32,11 +32,22 @@
...
+			gpio-export,name = "usb";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: "USB power" or something similar, more explanative :-)

File: target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts [hunk 3]
```diff
@@ -32,11 +32,22 @@
...
+			gpio-export,output = <1>;
+			gpios = <&gpio1 7 GPIO_ACTIVE_HIGH>;
+		};
+	};
 };
 
 &state_default {
 	gpio {
-		ralink,group = "i2c", "refclk", "wdt", "p3led_an", "wled_an";
+		ralink,group = "i2c", "refclk", "wdt", "p2led_an", "p3led_an", "wled_an";
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: p4led_an?

Comment (adrianschmutzler) [reply target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: Err, gpio1 7 should be 39? Or is it gpio1 9 = 41 as you state in commit message?

Comment (dengqf6) [reply target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: `GPIO1 7` is 41.

Comment (adrianschmutzler) [reply target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: So, gpio1 0 is 34?

Comment (dengqf6) [reply target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: Sorry, it should be 9, my bad :/

Comment (adrianschmutzler) [reply target/linux/ramips/dts/mt7628an_hiwifi_hc5761a.dts]: Did you retest with the DTS changes? Or just tried 41 on device?

**Comment by adrianschmutzler** [comment]
I suppose this can be backported down to 18.06?

**Comment by adrianschmutzler** [comment]
I've merged the switch patch for now, still waiting on feedback concerning the testing of the USB power.

**Comment by dengqf6** [comment]
I found that GPIO 41 should be low to enable GPIO power.
So should I use
```
gpio-export,output = <0>;
gpios = <&gpio1 9 GPIO_ACTIVE_HIGH>;
```
or
```
gpio-export,output = <1>;
gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
```
?

**Comment by dengqf6** [comment]
Found that `gpio-export,output = <0>;` set output to low regardless of `GPIO_ACTIVE_HIGH/GPIO_ACTIVE_LOW`

So I used
```
gpio-export,output = <0>;
gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
```

**Comment by dengqf6** [comment]
@adrianschmutzler ping

**Comment by adrianschmutzler** [comment]
@dengqf6 See this explanation (though on gpio-hog):
https://github.com/openwrt/openwrt/pull/2466#discussion_r331752746

So as you reported, it looks like you latest implementation is correct (I'm also always confused with these options here).

**Comment by dengqf6** [comment]
@adrianschmutzler 
gpio-hog does not allow users to control the port power
see openwrt/openwrt@b1e8a390ea7bde59eb47bdd729228cb377aa6f6f

**Comment by adrianschmutzler** [comment]
> gpio-hog does not allow users to control the port power

I'm aware of that, but the logic of setting the output vs. setting GPIO_OUTPUT_x consumer flag appeared to be the same, so I was just providing the link as FYI.

This is merged already BTW, thanks!

**Pull request closed without merging**